pycaret/internal/plots/time_series.py

```diff
@@ -812,18 +812,12 @@ def plot_time_series_decomposition(
     decomp_result = None
     data_ = data.to_timestamp() if isinstance(data.index, pd.PeriodIndex) else data
 
-    sp_to_use = data_kwargs.get("sp_to_use", None)
+    sp_to_use = data_kwargs.get("sp_to_use")
     if plot == "decomp_classical":
-        if sp_to_use is None:
-            decomp_result = seasonal_decompose(data_, model=classical_decomp_type)
-        else:
             decomp_result = seasonal_decompose(
```

Comment:
Could you please confirm that this works when `sp_to_use` = 1. If so, then this looks good. Thanks!

Reply:
The thing here is that `sp_to_use` cannot be None. So L818 was unreachable code really.

`seasonal_decompose` does work when `sp_to_use`, although the result is useless: a classical decomposition with no seasonality is all trend. Here is the classical decomposition for the airline data, if we impose `seasonal_period=1` and thus `sp_to_use=1``: 
![classical_sp_1](https://user-images.githubusercontent.com/8227377/151696741-221aa08f-45ca-4c50-a93d-ef607ffd7271.png)